src/components/CourseForm/CourseForm.jsx

```diff
@@ -44,12 +63,25 @@ export const CreateCourse = ({
 		setDescription(value);
 	};
 
-	const handleCreateCourse = () => {
+	const handleSaveCourse = () => {
 		if (!minutes || !title || !description || !selectedAuthors.length) {
 			return alert('Please, fill in all fields');
 		}
-		createNewCourse(title, description, minutes, selectedAuthors);
+		const courseData = {
+			title,
+			description,
+			creationDate: getCurrentDate(),
+			authors: selectedAuthors,
+			duration: minutes,
+		};
+		let action = createCourse;
+		if (courseId) {
+			action = updateCourseThunk;
+			courseData.id = courseId;
+		}
+		dispatch(action(courseData));
```

Comment:
It's really bad to reassign variables like this, especially with functions, why not do 
```
if (something) {
  dispatch(firstAction())
} else {
  dispatch(secondAction())
}
```
?